### ISSUE TITLE
Improved search in "Go to file" panel

### ISSUE THREAD

**Issue by alexandernanberg** [Microsoft/vscode#25925]
Would love to see a more accurate search for what I'm actually searching for. For example when I'm searching for "archive" and I'am in mid sentence i get these results:

![image](https://cloud.githubusercontent.com/assets/8997319/25721150/bc83aa46-310f-11e7-9edf-6ce228bd7ec4.png)

I think, at least for me that the index.js inside the archive folder file is much more relevant than the .babelrc and yarn.lock. Sublime on the other hand does exactly what I want. 

![image](https://cloud.githubusercontent.com/assets/8997319/25721267/16265666-3110-11e7-8bda-1f630b3f81f6.png)

This change would make the workflow with VS Code even better than it already is 😄

**Comment by chrmarti** [comment]
We use the path only as tiebreaker, that makes filenames with non-consecutive matches win against paths with consecutive matches here.

**Comment by bpasero** [comment]
Also https://github.com/Microsoft/vscode/issues/35572#issuecomment-339282965

**Comment by grothkopp** [comment]
Thank you @bpasero.
I like to add that lots of web frameworks like symfony have lots of files with identical names in different directories (indexSuccess.php, etc).

**Comment by bpasero** [comment]
@grothkopp this is only an issue if your query happens to fully match on the file name. once your query has enough significant characters from the parent folders it should not happen anymore.

**Comment by grothkopp** [comment]
@bpasero yes, but as my screenshot shows its easy to fully match the individual characters in long file names.

And partial matches in filesnames still have precedence (see my screeshot: **o**ne**sho**tRecurringTask.class.**p**hp in **li**b/task ranks before _betrag.ph**p** in **liosho**p)

There's also the issue with "greedyness" (the last p in php matching instead of the p in the path).

**Comment by grothkopp** [comment]
I just noticed that adding a trainling slash after the directory name ( "lioshop/") solves the problem in the insider build. 

But I think thats more a workaround than a solution.

**Comment by bpasero** [comment]
@grothkopp yes good point, if you include a path separator, the matching on the file name alone will no longer happen.

**Comment by ryenus** [comment]
@chrmarti 

Possible to make this a priority? locate a file using a relative path is much needed than only being able to find by just the name. Especially for people working with large projects with lots of modules/files.

Thanks

**Comment by chrmarti** [comment]
I would like to see us making progress here, but don't have the necessary time at the moment. Some care will be needed, because with large projects the sorting affects performance.

**Comment by EmmaSimon** [comment]
I'm not sure if this should be a different issue, but I'd also like to see independent fuzzy finding for multiple words. This was added in sublime recently, and I miss it in other editors.
https://forum.sublimetext.com/t/dev-build-3154/33286
> Also changed in 3154 is the fuzzy matching algorithm (as used by the Command Palette and Goto Anything), where you can now enter space separated terms, and each will be matched independently. e.g., "hw" will match "Hello World", and now "w h" will also match it (but "wh" would not).

For a more specific example, I have some files in a `/reducers` directory, and whenever I want one I usually type `user reducer`, and since `reducer` is before `user` in the file path, the file isn't found.

**Comment by bpasero** [comment]
Yes, https://github.com/microsoft/vscode/issues/25925#issuecomment-357261679 should be covered with https://github.com/microsoft/vscode/issues/30404

**Comment by garygreen** [comment]
Any progress on this? It's been 3 years and Sublime still outclasses VSCode when it comes to fuzzy search matching in a intuitive way. Would be super awesome if more focus could be made on improving this. I think a lot of developers would appreciate it 👍

**Comment by ses4j** [comment]
Is it possible to write an extension to replace the file search provider?  I agree with many previous commenters that Sublime's gets it right so much more often than VS Code's... i really miss it.  One piece of the algorithm that I think is sorely missing is that a search like `hw` should match `hello_world.py` better than it matches `howto.py`, by recognizing that underscores (or camelcase, etc) are natural word breaks and first letters of words are more important than others...

**Comment by mattaningram** [comment]
Another example screenshot showing that path names should not be used just as tie breakers. Path name exact string matches should appear higher than simply finding all the letters somewhere in the filename. This is just confusing:

![image](https://user-images.githubusercontent.com/955092/98388312-9a051a00-2020-11eb-915f-486ec58664ce.png)

**Comment by pforhan** [comment]
Is there a separate issue about sorting case-sensitive matches higher?  Let me know.
For example, I'd like to see HelpAppletClientActionTranslatorTest bubble up to the top here:
![image](https://user-images.githubusercontent.com/331925/99426610-ecb8be80-28c9-11eb-84dd-86cbff303f78.png)

**Comment by rcywongaa** [comment]
Since users are most likely jumping to files related to the code they are currently editing, we could borrow some ideas from https://github.com/microsoft/vscode/issues/80444, in particular, order by proximity to currently editing file would seem like a low hanging fruit

I've had great success using fzf with [proximity-sort](https://github.com/jonhoo/proximity-sort)

**Comment by JUSTIVE** [comment]
I'm having similar needs as @pforhan .
<img width="495" alt="image" src="https://github.com/user-attachments/assets/f9ebb913-fabc-4213-9284-387ba2dad15c">
I want to access my files with their initial letters, not just any matching files

**Comment by mrmachine** [comment]
Bringing my comments from the closed as duplicate issue above so they can add to the conversation.

It's not only Sublime Text that gets it right. GitHub does a way better job and basically does pretty much exactly what I expect every time I use it.

Surely it should not be too hard for MS to copy the implementation from GH, which has VS Code baked into it, which they also own, and which runs in a web browser without performance issues?

It is disheartening to see that this issue is still open after nearly 7 and a half years.

Original comments below:

I always have terrible results with the VS Code Quick Open feature, and excellent results with the GitHub "go to file" feature.

For example, I want to jump to a file `djangosite/urls/default.py` so I Quick Open and type `urldef`, but I get this soup in VS Code:

<img width="612" alt="image" src="https://github.com/user-attachments/assets/d41e239a-ad5a-4a6a-8e64-72b56b06edb4">

In GitHub with "go to file", I get exactly what I want:

<img width="645" alt="image" src="https://github.com/user-attachments/assets/2c8c608e-092c-417e-b4cf-b465adb12fea">

If I try to separate the two fragments with a space `url def` (and even try reversing them to `def url` thinking the first fragment might better match the filename and then be narrowed by path), I get better results, but still the best match is 4th down on the list below what appear to be lower quality matches:

<img width="613" alt="image" src="https://github.com/user-attachments/assets/5f1f4595-9c53-4b2c-b5ab-bc6f842362e1">
<img width="613" alt="image" src="https://github.com/user-attachments/assets/67236f52-8fa2-4dd5-b1c2-b70834b94bc3">

Why does a 3 consecutive character match from the start of the parent directory and covering 75% of the parent directory name (`url`s) not trump 3x single character matches from anywhere inside separate path fragments? The individually matching characters make up a tiny percentage of the overall length of each fragment, and some are not even matching anywhere near the start of the fragment.

This is my experience often with VS Code where I have to ignore and visually scan past a long list of garbage matches to find the one I actually want. Which is infuriating with the GitHub implementation nails it pretty much every time.

**Comment by bpasero** [comment]
This is a good example, thanks for sharing. In both cases our algorithm prefers a better performing match in the file name over a match in the full path and that is the root cause of many issues. Addressing this would mean to give up on the separation between file name matches and path matches and focus only on path matches I think.

**Comment by mrmachine** [comment]
I would also suggest that in response to your comment about the algorithm preferring a better match in the filename, that these "better" filename matches shown in my earlier screenshots are not actually better at all considering that pretty much ALL of the top matches include single isolated character matches in the middle of a word in the filename which is extremely low value.

**Comment by lhnrd** [comment]
Any news about this feature? I'd love to see a plug n play kind of option to improve the search. Often I can't find the file in a big project.
